packages/seda-chain-contracts/src/utils.rs

```diff
@@ -24,3 +25,42 @@ pub fn check_eligibility(deps: &DepsMut, dr_executor: Addr) -> Result<bool, Cont
     let is_eligible = ELIGIBLE_DATA_REQUEST_EXECUTORS.load(deps.storage, dr_executor)?;
     Ok(is_eligible)
 }
+
+pub fn get_attached_funds(funds: &[Coin], token: String) -> Result<u128, ContractError> {
+    let amount: Option<u128> = funds
+        .iter()
+        .find(|coin| coin.denom == token)
+        .map(|coin| coin.amount.u128());
+    amount.ok_or(ContractError::NoFunds)
+}
+
+pub fn pad_to_32_bytes(value: u128) -> [u8; 32] {
+    let mut bytes = [0u8; 32];
+    let small_bytes = &value.to_be_bytes();
+    bytes[(32 - small_bytes.len())..].copy_from_slice(small_bytes);
+    bytes
+}
+
+pub fn hash_update(hasher: &mut Keccak256, value: u128) {
+    let bytes = pad_to_32_bytes(value);
+    hasher.update(bytes);
+}
+
+pub fn hash_data_request(
+    nonce: u128,
+    value: String,
```

Comment:
I believe this could be `&str` as well since `&str` also implements `as_bytes` I think. Might be wrong about that one.

Reply:
Both vecs could also be passed as 1`&[u8]` instead!

Reply:
Not sure if it's worth it for `wasm_args` since it adds an extra cast in `post_data_request`:
```
        let wasm_args_slice: Vec<&[u8]> = wasm_args.iter().map(AsRef::as_ref).collect();
        let reconstructed_dr_id =
            hash_data_request(&nonce, &value, &chain_id, &wasm_id, &wasm_args_slice);
```